db.py

```diff
@@ -172,6 +173,51 @@ class CourseOrganization(db.Model):
     )
 
 
+class Assignment(db.Model):
+    __tablename__ = 'assignment'
+    id = db.Column(db.Integer, primary_key=True)
+    is_draft = db.Column(db.Boolean, default=True)
+
+    assignment_lines = db.relationship('AssignmentLine', backref='assignments', lazy=True)
+    comments = db.relationship('Comment', backref='comments', lazy=True)
+    users = db.relationship('User',
+                            secondary='assignment_line',
+                            primaryjoin="Assignment.id == AssignmentLine.assignment_id",
+                            secondaryjoin="AssignmentLine.user_id == User.id",
+                            backref='assignments',
+                            lazy=True)
+
+
+class AssignmentLine(db.Model):
+    __tablename__ = 'assignment_line'
+    id = db.Column(db.Integer, primary_key=True)
+    course_id = db.Column(db.Integer, nullable=False)
+    course_year = db.Column(db.Integer, nullable=False)
+    user_id = db.Column(db.Integer, db.ForeignKey('user.id'))
+    load_q1 = db.Column(db.Integer)
+    load_q2 = db.Column(db.Integer)
+    position = db.Column(db.Integer)
+    assignment_id = db.Column(db.Integer, db.ForeignKey('assignment.id'))
+
+    __table_args__ = (
+        db.ForeignKeyConstraint(
+            ['course_id', 'course_year'],
+            ['course.id', 'course.year']
+        ),
+    )
+
+
+class Comment(db.Model):
+    __tablename__ = 'comment'
+
+    id = db.Column(db.Integer, primary_key=True)
+    row = db.Column(db.Integer, nullable=False)
```

Comment:
I'm totally confused here. Can't we simply associate the comment to the assignment, which corresponds to a cell in practice ?

Saving cell column and line won't work if the dataset change (courses or researchers).

Reply:
If there is only one comment per assignment then we should definitely have it stored in the assignment entry. 

Yes, the comment will have to be generated using the row and column info at the moment we generate the table.
You should always design an app with keeping in mind the fact data and data structure can change.